### PR TITLE
[POC] Spacing with CssValueInput

### PR DESCRIPTION
**Pull request by kof**
## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [ ] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx [hunk 1]
```diff
@@ -111,61 +120,87 @@ const styles = {
...
 const TextField = ({ property, value, onEnter }: TextFieldProps) => {
-  const [currentValue, setCurrentValue] = useState<string>(value ?? "");
+  const [currentValue, setCurrentValue] = useState<UnitValue>(
+    value ?? initialValue
+  );
   const isFromCurrentBreakpoint = useIsFromCurrentBreakpoint(property);
 
   useEffect(() => {
-    setCurrentValue(value ?? "");
+    setCurrentValue(value ?? initialValue);
```
Comment (istarkov) [inline apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx:145]: Usually you don't need useEffect here [explanation](https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes)

File: apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx [hunk 2]
```diff
@@ -111,61 +120,87 @@ const styles = {
...
+          css={{
+            background: "none",
+            boxShadow: "none",
+            borderRadius: 0,
+            minWidth: 80,
+            width: 80,
+            padding: 2,
+            animationDuration: "200ms",
+            animationTimingFunction: "cubic-bezier(0.16, 1, 0.3, 1)",
+            willChange: "transform, opacity",
```
Comment (istarkov) [inline apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx:176]: willChange  usually has no effect or even degrades perf.  Also doc `Warning: will-change is intended to be used as a last resort, in order to try to deal with existing performance problems. It should not be used to anticipate performance problems.`

Here depends on Popover implementation, as setting it with element creation usually has no effect. Should be set before animation begins. Anway somekind of magic.

File: apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx (hunk 1 above)
Comment (kof) [reply apps/designer/app/designer/features/style-panel/sections/spacing/spacing.tsx:145]: good catch, didn't know this one

**Comment by kof** [comment]
@rpominov is this still useful or should I close?

**Comment by rpominov** [comment]
Not going to merge, but may need the branch

**Pull request closed without merging**